lib/widgets/message_list.dart

```diff
@@ -511,60 +533,93 @@ class _UnreadMarker extends StatelessWidget {
   }
 }
 
-class StreamTopicRecipientHeader extends StatelessWidget {
-  const StreamTopicRecipientHeader({super.key, required this.message});
+class StreamMessageRecipientHeader extends StatelessWidget {
+  const StreamMessageRecipientHeader({super.key, required this.message, required this.showStream});
 
   final StreamMessage message;
+  final bool showStream;
 
   @override
   Widget build(BuildContext context) {
+    // For design specs, see:
+    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=538%3A20849&mode=dev
+    //   https://github.com/zulip/zulip-mobile/issues/5511
     final store = PerAccountStoreWidget.of(context);
 
     final stream = store.streams[message.streamId];
-    final streamName = stream?.name ?? message.displayRecipient; // TODO(log) if missing
     final topic = message.subject;
-
-    final subscription = store.subscriptions[message.streamId];
-    final streamColor = Color(subscription?.color ?? 0x00c2c2c2);
+    final swatch = store.subscriptions[message.streamId]?.colorSwatch()
+      ?? _kFallbackStreamColorSwatch;
     final contrastingColor =
-      ThemeData.estimateBrightnessForColor(streamColor) == Brightness.dark
+      (ThemeData.estimateBrightnessForColor(swatch.barBackground) == Brightness.dark)
         ? Colors.white
         : Colors.black;
+    final textStyle = TextStyle(
+      color: contrastingColor,
+      fontFamily: 'Source Sans 3',
+      fontSize: 16,
+      letterSpacing: 0.02 * 16,
+      height: (18 / 16),
+    ).merge(weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900));
+
+    final streamWidget = (showStream)
+      ? GestureDetector(
+          onTap: () => Navigator.push(context,
+            MessageListPage.buildRoute(context: context,
+              narrow: StreamNarrow(message.streamId))),
+          child: Row(children: [
+            Padding(
+              // Figma shows 5px spacing but 6px here matches better visually.
+              padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 11),
```

Comment:
Hmm interesting. What does it make a better match with — does it better match the result to what you see in Figma, or match this to another part of the design, or something else?

Reply:
Curious. I wonder what the discrepancy is, then — in principle if we match the numbers in the Figma spec, we should get the same visual results as the Figma preview.

Given the Figma bug we [just discovered](https://github.com/zulip/zulip-flutter/pull/416#discussion_r1406723537) about opacity, it wouldn't be shocking if this was a bug in Figma where its preview didn't interpret its spec correctly. But it could also be a bug in our translation of the spec.

Reply:
On further examination here it looks like the icon is given a size of 18px but the icon itself has width/height of 16px here (there is 1px of space inside the icon!) and so there are 6 pixels of whitespace from the left border:
![figma_stream_icon](https://github.com/zulip/zulip-flutter/assets/98299/30a07386-94d9-4180-a1a1-8b710fab8bc0)

Another result of this examination is I adjusted the icon size to be 16px instead.